bug: arrays not being returned correctly

UPDATE: arrays of any type (int, string) are not being parsed correctly for any source format: JSON, TOML, and YAML.

When trying to populate a flag containing a string array coming from a TOML file, I discovered that instead of returning a slice with one string per array item, instead it is returning a slice with a single string that contains the entire array.

I added a unit test that demonstrates the problem in a fork/branch:

```shell
$ go test ./...
ok      github.com/urfave/cli-altsrc/v3 (cached)
ok      github.com/urfave/cli-altsrc/v3/json    (cached)
--- FAIL: TestTOMLArray (0.00s)
    toml_value_source_test.go:67: 
                Error Trace:    /home/ron/Development/cli-altsrc/toml/toml_value_source_test.go:67
                Error:          Not equal: 
                                expected: []string([]string{"hello", "hi", "hola", "hej"})
                                actual  : string("[hi hello hola hej]")
                Test:           TestTOMLArray
FAIL
FAIL    github.com/urfave/cli-altsrc/v3/toml    0.008s
ok      github.com/urfave/cli-altsrc/v3/yaml    (cached)
FAIL
```

Here is a commit with the failing test:
https://github.com/hybridgroup/cli-altsrc/commit/8d09ba699f6aedf859869a37c0c576552980c612


Thanks for any help!

--- a/.testdata/slice.json
+++ b/.testdata/slice.json
@@ -1,0 +1,13 @@
+{
+  "slice_types": {
+    "numbers": [
+      1,
+      2,
+      3
+    ],
+    "strings": [
+      "word",
+      "this is a sentence"
+    ]
+  }
+}

--- a/.testdata/slice.toml
+++ b/.testdata/slice.toml
@@ -1,0 +1,10 @@
+[slice_types]
+numbers = [
+  1,
+  2,
+  3,
+]
+strings = [
+  "word",
+  "this is a sentence",
+]

--- a/.testdata/slice.yaml
+++ b/.testdata/slice.yaml
@@ -1,0 +1,8 @@
+slice_types:
+  numbers:
+    - 1
+    - 2
+    - 3
+  strings:
+    - "word"
+    - "this is a sentence"

--- a/altsrc.go
+++ b/altsrc.go
@@ -4,6 +4,7 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"reflect"
 	"runtime"
 	"strings"
 )
@@ -115,6 +116,23 @@ type ValueSource struct {
 func (vs *ValueSource) Lookup() (string, bool) {
 	maafsc := NewMapAnyAnyURISourceCache(vs.sourcer.SourceURI(), vs.um)
 	if v, ok := NestedVal(vs.key, maafsc.Get()); ok {
+		// Use reflection to check if 'v' is a slice
+		if val := reflect.ValueOf(v); val.Kind() == reflect.Slice {
+			// It's a slice, so create a new string slice of the same size
+			stringSlice := make([]string, val.Len())
+
+			// Iterate over the slice elements
+			for i := 0; i < val.Len(); i++ {
+				// Get the element at index i and convert it to a string
+				elem := val.Index(i).Interface()
+				stringSlice[i] = fmt.Sprintf("%v", elem)
+			}
+
+			// Join the string representations and return
+			return strings.Join(stringSlice, ","), true
+		}
+
+		// Fall back to standard string representation if not a slice
 		return fmt.Sprintf("%[1]v", v), ok
 	}
 

--- a/json/json_value_source_test.go
+++ b/json/json_value_source_test.go
@@ -50,3 +50,35 @@ func TestJSON(t *testing.T) {
 	r.Equal("yaml file \"/dev/null/nonexistent.json\" at key \"water_fountain.water\"", yvs.String())
 	r.Equal("yamlValueSource{file:\"/dev/null/nonexistent.json\",keyPath:\"water_fountain.water\"}", yvs.GoString())
 }
+
+func TestJSONSlice(t *testing.T) {
+	r := require.New(t)
+
+	configPath := filepath.Join(testdataDir, "slice.json")
+
+	t.Run("numbers", func(t *testing.T) {
+		vs := JSON(
+			"slice_types.numbers",
+			altsrc.StringSourcer(configPath),
+		)
+
+		vsc := cli.NewValueSourceChain(vs)
+
+		v, ok := vsc.Lookup()
+		r.Equal("1,2,3", v)
+		r.True(ok)
+	})
+
+	t.Run("strings", func(t *testing.T) {
+		vs := JSON(
+			"slice_types.strings",
+			altsrc.StringSourcer(configPath),
+		)
+
+		vsc := cli.NewValueSourceChain(vs)
+
+		v, ok := vsc.Lookup()
+		r.Equal("word,this is a sentence", v)
+		r.True(ok)
+	})
+}

--- a/toml/toml_value_source_test.go
+++ b/toml/toml_value_source_test.go
@@ -50,3 +50,35 @@ func TestTOML(t *testing.T) {
 	r.Equal("toml file \"/dev/null/nonexistent.toml\" at key \"water_fountain.water\"", tvs.String())
 	r.Equal("tomlValueSource{file:\"/dev/null/nonexistent.toml\",keyPath:\"water_fountain.water\"}", tvs.GoString())
 }
+
+func TestTOMLSlice(t *testing.T) {
+	r := require.New(t)
+
+	configPath := filepath.Join(testdataDir, "slice.toml")
+
+	t.Run("numbers", func(t *testing.T) {
+		vs := TOML(
+			"slice_types.numbers",
+			altsrc.StringSourcer(configPath),
+		)
+
+		vsc := cli.NewValueSourceChain(vs)
+
+		v, ok := vsc.Lookup()
+		r.Equal("1,2,3", v)
+		r.True(ok)
+	})
+
+	t.Run("strings", func(t *testing.T) {
+		vs := TOML(
+			"slice_types.strings",
+			altsrc.StringSourcer(configPath),
+		)
+
+		vsc := cli.NewValueSourceChain(vs)
+
+		v, ok := vsc.Lookup()
+		r.Equal("word,this is a sentence", v)
+		r.True(ok)
+	})
+}

--- a/yaml/yaml_value_source_test.go
+++ b/yaml/yaml_value_source_test.go
@@ -49,3 +49,35 @@ func TestYAML(t *testing.T) {
 	r.Equal("yaml file \"/dev/null/nonexistent.yaml\" at key \"water_fountain.water\"", yvs.String())
 	r.Equal("yamlValueSource{file:\"/dev/null/nonexistent.yaml\",keyPath:\"water_fountain.water\"}", yvs.GoString())
 }
+
+func TestYAMLSlice(t *testing.T) {
+	r := require.New(t)
+
+	configPath := filepath.Join(testdataDir, "slice.yaml")
+
+	t.Run("numbers", func(t *testing.T) {
+		vs := YAML(
+			"slice_types.numbers",
+			altsrc.StringSourcer(configPath),
+		)
+
+		vsc := cli.NewValueSourceChain(vs)
+
+		v, ok := vsc.Lookup()
+		r.Equal("1,2,3", v)
+		r.True(ok)
+	})
+
+	t.Run("strings", func(t *testing.T) {
+		vs := YAML(
+			"slice_types.strings",
+			altsrc.StringSourcer(configPath),
+		)
+
+		vsc := cli.NewValueSourceChain(vs)
+
+		v, ok := vsc.Lookup()
+		r.Equal("word,this is a sentence", v)
+		r.True(ok)
+	})
+}

